src/asar-utils.ts

```diff
@@ -31,3 +58,153 @@ export const generateAsarIntegrity = (asarPath: string) => {
       .digest('hex'),
   };
 };
+
+function toRelativePath(file: string): string {
+  return file.replace(/^\//, '');
+}
+
+function isDirectory(a: string, file: string): boolean {
+  return Boolean('files' in asar.statFile(a, file));
+}
+
+function checkUnique(archive: string, file: string, allowList?: string): void {
+  if (allowList === undefined || !minimatch(file, allowList, { matchBase: true })) {
+    throw new Error(
+      `Detected unique file "${file}" in "${archive}" not covered by ` +
+        `allowList rule: "${allowList}"`,
+    );
+  }
+}
+
+export const fuseASARs = async ({
+  x64,
+  arm64,
+  output,
```

Comment:
Can these all be more specific, I assume they're paths to asar files?

So `x64AsarPath`, `arm64AsarPath`, `outputAsarPath`, etc.